Aquarius/Src/Aquarius/Renderer/Texture.cpp

```diff
@@ -0,0 +1,61 @@
+#include "Texture.h"
+
+
+namespace Aquarius {
+
+	Texture::Texture(const std::string& texture, TextureConfiguration configuration, bool hasAlpha)
+		: m_Configuration(configuration)
+	{
+		generateTexture(texture, hasAlpha);
+	}
+
+	bool Texture::generateTexture(const std::string& texture, bool hasAlpha)
+	{
+		bool success = false;
+		stbi_set_flip_vertically_on_load(true);
+
+		glGenTextures(1, &m_ID);
+		glBindTexture(GL_TEXTURE_2D, m_ID);
+
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, static_cast<int>(m_Configuration.textureWrapS));
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, static_cast<int>(m_Configuration.textureWrapT));
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, static_cast<int>(m_Configuration.minTextureFiltering));
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, static_cast<int>(m_Configuration.magTextureFiltering));
+		unsigned char* data = stbi_load(texture.c_str(), &m_Width, &m_Height, &m_NumColorChannels, 0);
+		
+		if (data != nullptr)
+		{
+			uint32_t format = hasAlpha ? GL_RGBA : GL_RGB;
+
+			glTexImage2D(GL_TEXTURE_2D, 0, format, m_Width, m_Height, 0, format, GL_UNSIGNED_BYTE, data);
```

Comment:
Technically the first format argument is the internal format and should be RGBA8 or RGB8 (RGBA works only for backwards compatibility according to youtube)

Reply:
Ahh I knew they were different but in learnopenGL they used GL_RGB and RGBA for both, but this is an easy fix